### PR TITLE
Support webview ui variant

### PR DESCRIPTION
**Pull request by carmenlau**
ref #111

### CODE REVIEW

File: sdk/src/main/java/com/oursky/authgear/UIVariant.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.oursky.authgear
+
+enum class UIVariant {
+    CustomTabs,
+    WebView
```
[the remark this answers is not in the document]
Comment (carmenlau) [reply sdk/src/main/java/com/oursky/authgear/UIVariant.kt:5]: We are using camelcase on the Page and ColorScheme enum... Let me change it as well

**Comment by carmenlau** [comment]
Today we have discussed we want to avoid unnecessary JS calls in the full-screen mode to make the solution more generic. 
1. We will only handle auto-cancel when the HTML is failed to load.
2. If the user clicks cancel from the UI, we call the redirect URL with the error cancel.

So we plan to not add any js for `onLoaded` or `goBack`. 

@kiootic Would like to know:
1. Is it okay for this plan?
2. From the android webview POC, I saw  `updateSafeArea` is called after loaded. Is it necessary to be called after the JS is loaded? Or can we call it after the HTML is loaded (`onPageFinished` of `WebViewClient`)? 

cc @louischan-oursky

**Comment by kiootic** [comment]
> Is it okay for this plan?

If the CSS/JS failed to load, the full-screen variant would got stuck in white screen/error screen. Is that ok?

> From the android webview POC, I saw updateSafeArea is called after loaded. Is it necessary to be called after the JS is loaded? Or can we call it after the HTML is loaded (onPageFinished of WebViewClient)?

It's because the injected JS would get null `document.documentElement` if the page is not yet loaded sufficiently. I think using `onPageCommitVisible` for the flag should be ok, let me update the POC with this.

**Comment by louischan-oursky** [comment]
@carmenlau @kiootic The keyboard handling has some problem. See the attached video.


https://user-images.githubusercontent.com/15082564/211452666-394df773-326e-4b91-8a26-83115b3c267b.mp4

Before the keyboard appears, the webview is scrollable. After the keyboard appeared, the webview is not scrollable.

**Comment by carmenlau** [comment]
Updated the keyboard layout handing from the POC

I kept the `updateSafeArea` javascript. I think we still need them, right? @kiootic 

https://user-images.githubusercontent.com/388892/212127134-6efdb1f2-7774-473c-b76d-462f9cbeed9c.mov

**Comment by kiootic** [comment]
We're putting the web view within the safe area on Android, so there should be no need to keep that.

**Comment by louischan-oursky** [comment]
@kiootic Does my last commit look good? It removed safe area handling.

**Comment by kiootic** [comment]
Yes looks good to me.

**Pull request merged**